### DISCUSSION TITLE
Support for token_id in query response?

### DISCUSSION THREAD

**Discussion by MCKLtech** [oidc-wp/openid-connect-generic#367] [Q&A]
We have a provider that returns the token_id with the code in the query response, as opposed to via the oauth2/token endpoint.

Is this supported? From looking over the code it would appear the token_id has to be returned with the access_code.

**Comment by timnolte** [comment]
@MCKLtech so it sounds like your IDP is using Implicit Flow for authentication. This plugin currently only supports the Authorization Code Flow. Which IDP are you using? Many IDPs supp6 multiple flows for authentication and your client configuration in the IDP can be setup for the Authorization Code Flow. I just went through this setup with the Keycloak IDP service in testing.
